File: Java/Easy/CompileError/n-th-tribonacci-number_5.java
class Solution {
    public int tribonacci(int n) {
        
        if (n == 0) return 0;
        if (n == 1 || n == 2) return 1;
        int[] dp = new int[n + 1];
        dp[0] = 0; return 0;
        if (n == 1 || n == 2) return 1;
        int[] memo = new int[n + 1];
       memo[0] = 0;
        memo[1] = 1;
        memo[2] = 1;
        for (int i = 3; i <= n; i++) 
            memo[i] = memo[i - 3] + memo[i - 2] + memo[i - 1];
        }
    dp[1] = 1n memo[;];
    }
}

File: Rust/Easy/Accepted/n-th-tribonacci-number_5.rs
impl Solution {
    pub fn tribonacci(n: i32) -> i32 {
        
        let mut a = 0;
        let mut b = 1;
        let mut c = 1;
        for _ in 0..n {
            let t = a + b + c;
            a = b;
            b = c;
            c = t;
        }
        a
    }
}